### ISSUE TITLE
Scrolling video not fix the current video

### ISSUE THREAD

**Issue by hafiz013** [eneim/toro#335]
What kind of issue is this?
when i click on those list video and come back scrolling the video position which are previously on third row move to first row and first row goes to third row

- [ ] Question. If so, declare clearly your **target** and your current implementation with code formatted in markdown.
refer attachment code
[VideoListExo.zip](https://github.com/eneim/toro/files/2020947/VideoListExo.zip)

- [ ] Bug report. If so, please supply the way to reproduce the issue, including addition information as follow:
  - [ ] Toro version.
im.ene.toro3:toro:3.4.1, im.ene.toro3:toro-ext-exoplayer:3.4.1, com.google.android.exoplayer:exoplayer:2.7.3
  - [ ] Android version (can be found in Device info). **Rooted/Custom rom devices are not supported**.
android 6 above
  - [ ] Device name (eg: Nexus 5X, Samsung Galaxy S8)
xiomi
  - [ ] Source code that can reproduce the issue (a snapshot is expected).
refer to attachment.
  - [ ] Logcat stack trace in case of fatal issue.
 don't have.

- [ ] Feature request. Start by telling me what problem you’re trying to solve. Don’t send pull requests to implement new features without first getting my support. Sometimes I leave features out on purpose to keep the project small.
The feature of this project be able to to show full screen of video when click enlarge button. However when i click back the shrink button the video position of first row goes to third row and third row video go to first row.
 
- [ ] Other. Senseless issues are normally ignored. Please understand and cooperate.
nope

**Comment by eneim** [comment]
@hafiz013 Just have a quick look over your source code. One point is your implementation of ViewHolder and full-screen dialog doesn't take into account the 'recycle' characteristic of ViewHolder. But let me go deep into your code to make a better understanding of your use-case and think about some suggestion as well as improvement to my work.

**Comment by hafiz013** [comment]
yes sure because if you could solve my problem the other also use my suggestion as well. this is because the point of list is that user can click full screen of the video (in landscape) and shrink back (in portrait). Secondly what is the different between contain and recyclerview?

**Comment by eneim** [comment]
@hafiz013 Container is a RecyclerView but has been empowered. You can see that Container extends RecyclerView.

**Comment by eneim** [comment]
@hafiz013 So I just try to run your code and here is what I found:

**SimpleAdapter.java** line 136 (in ``closeFullscreenDialog()``):

```java
((FrameLayout) mContext.findViewById(R.id.main_media_frame)).addView(playerView);
```

Here you use mContext from Activity to find R.id.main_media_frame. In fact, RecyclerView holds 3 view that contains ``R.id.main_media_frame`` as its child. So using ``mContext.findViewById(R.id.main_media_frame)`` will find the first one, which is the view on top. That's why you see **the video position of first row goes to third row and third row video go to first row.**.

Changing ``mContext`` to ``itemView.getContext()`` will fix it.

In fact, this is not an issue of Toro. Please take a look :D.

**Comment by hafiz013** [comment]
can you do tutorial about video full screen in new post so that another user can refer to it

**Comment by hafiz013** [comment]
hi i still persistent to get landscape when full screen because this code still have bugs.

**Comment by eneim** [comment]
@hafiz013 About the tutorial, I have one sample and also I try to not be opinionated about how to do this and that. But this is a good idea. About land scape mode, is it the same behaviour? Or what doesn't work well this time? Because switching to landscape mode depends so much on your setup, I cannot tell anything now.

**Comment by hafiz013** [comment]
About land scape mode, is it the same behaviour?  nope. when landscape only one video play at the time and when go back shrink list video will appear

**Comment by eneim** [comment]
@hafiz013 Can you take a look at my Facebook demo? Using DialogFragment for full-screen player there. Tell me if it makes sense to you.

**Comment by hafiz013** [comment]
may I know what is your facebook page. thank you

**Comment by eneim** [comment]
@hafiz013 Do you mean Facebook demo? It is here: [link](https://github.com/eneim/toro/tree/dev-v3/app/src/main/java/im/ene/toro/sample/facebook).

If you want to ask for my personal Facebook then sorry ^^!

**Comment by syedmumersajjad** [comment]
@hafiz013 Assalam o alikum i am looking for the toro exoplayer integration but didn't find the right solution for it can you please help me little so i'll be on path thanks

**Comment by eneim** [comment]
@syedmumersajjad The "app" module contains many demos but each folder is isolated. You can clone the source code, cope the source of folder facebook and folder "common" (Base classes) only and it should be Ok. All the dependencies (in build.gradle) are not required, but please clean them up by yourself.

**Comment by syedmumersajjad** [comment]
@eneim thanks for your reply yeah it's work perfectly. I have implement all the demos separately Thanks a lot great work. I have one more question i didn't see the progress bar and caching video option. Can you help me little on these points?
